aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterInstrumentedTest.java

```diff
@@ -582,15 +582,52 @@ public void deleteModelCascades() throws DataStoreException {
         assertFalse(blogIterator.hasNext());
     }
 
+    /**
+     * Test delete with predicate. Conditional delete is useful for making sure that
+     * no data is removed with outdated assumptions.
+     * @throws DataStoreException from possible underlying DataStore exceptions
+     */
+    @Test
+    public void deleteModelWithPredicateDeletesConditionally() throws DataStoreException {
+        final BlogOwner john = BlogOwner.builder()
+                .name("John")
+                .build();
+        final BlogOwner jane = BlogOwner.builder()
+                .name("Jane")
+                .build();
+        final BlogOwner mark = BlogOwner.builder()
+                .name("Mark")
+                .build();
+        saveModel(john);
+        saveModel(jane);
+        saveModel(mark);
+
+        // Delete everybody but Mark
+        final QueryPredicate predicate = BlogOwner.NAME.ne(mark.getName());
+        deleteModel(john, predicate);
+        deleteModel(jane, predicate);
+        deleteModel(mark, predicate); // Should not be deleted
+
+        Iterator<BlogOwner> blogOwners = queryModel(BlogOwner.class);
+        assertNotNull(blogOwners);
+        assertEquals(mark, blogOwners.next());
```

Comment:
What guarantees that `mark` is the first result on the `Iterator`, here?

We should really rework `queryModel` so that it returns a `Set`/`List` directly and then we can just ask like `results.contains(mark)` and be done.

Reply:
good point.

Reply:
Thanks for making this change in another PR, @raphkim !